phd-tests/tests/src/hyperv.rs

```diff
@@ -0,0 +1,77 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use phd_framework::{artifacts, lifecycle::Action, TestVm};
+use phd_testcase::*;
+use tracing::warn;
+
+/// Attempts to see if the guest has detected Hyper-V support. This is
+/// best-effort, since not all PHD guest images contain in-box tools that
+/// display the current hypervisor vendor.
+async fn guest_detect_hyperv(vm: &TestVm) -> anyhow::Result<()> {
+    if vm.guest_os_kind().is_linux() {
+        // Many Linux distros come with systemd installed out of the box. On
+        // these distros, it's easiest to use `systemd-detect-virt` to determine
+        // whether the guest thinks it's running on a Hyper-V-compatible
+        // hypervisor. (Whether any actual enlightenments are enabled is another
+        // story, but those can often be detected by other means.)
+        let out = vm.run_shell_command("systemd-detect-virt").await?;
+        if out.contains("systemd-detect-virt: not found") {
+            warn!(
+                "guest doesn't support systemd-detect-virt, can't verify it \
+                detected Hyper-V support"
+            );
```

Comment:
Would it make sense to `phd-skip` out of the test in this case?

Reply:
I considered this but think it's useful for the smoke tests in this module to produce a "Pass" result to show that they successfully launched a guest in a Hyper-V-emulating Propolis, even if the guest didn't offer a means to detect the hypervisor type. (In an earlier version of this PR I had a bug that made the enlightenment stack panic on shutdown...) I've added a comment about this in c721e11e.

Reply:
Ah, that's a fair point, thanks for adding the comment!